### ISSUE TITLE
Missing Hyperparameter combinations on HPARAMS tab

### ISSUE THREAD

**Issue by N1h1l1sT** [pytorch/pytorch#62572]
## 🐛 Bug

When I use the SummaryWriter() function from torch.utils.tensorboard to create Tensorboard logs of the hyper-parameters and metrics (scalar), as well as the losses (image plot), and the model (graphs), only a subset of my hyper-parameter combinations are visible on the HPARAMS tab, however all are visible on the rest of the tabs (SCALAR, IMGAES, GRAPHS, TIME SERIES).

My python reprex below will create new random runs, so a different percentage of hparams will be visible on HPARAMS each time - it's usually between 20% and 40% of the runs.

## To Reproduce

1. Download and run my already created [Tensorboard logs](https://www.dropbox.com/s/9u1qev5rao2dg3p/Pytorch-Tensorboard-Missing-HParams.zip?dl=1), or
2. Download my [jupyter notebook](https://www.dropbox.com/s/uohmh6quo8tg190/HParams_missing_issue_reprex.ipynb?dl=1) to reproduce the problem, or
3. View the code as regular python
3.1 by [downloading the .py here](https://www.dropbox.com/s/mthklonqo5ft91t/HParams_missing_issue_reprex.py?dl=1), or
3.2 by viewing the python code from my dropbox from here: `https://www.dropbox.com/s/mthklonqo5ft91t/HParams_missing_issue_reprex.py?dl=0`

## Expected behavior

HPARAMS tab should display information from all the runs, like the rest of the tabs do.
This way, I can sort by 'test loss' on HPARAMS and see which hyper-parameter combination provided me with the best results.

## Environment

Collecting environment information...
PyTorch version: 1.9.0
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Pro
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.5 | packaged by conda-forge | (default, Jun 19 2021, 00:22:33) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19043-SP0
Is CUDA available: True
CUDA runtime version: 11.1.105
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 2080 SUPER
Nvidia driver version: 471.41
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.0
[pip3] pytorch-pretrained-bert==0.6.2
[pip3] torch==1.9.0
[pip3] torchaudio==0.9.0
[pip3] torchinfo==0.1.5
[pip3] torchtext==0.10.0
[pip3] torchvision==0.10.0
[pip3] torchviz==0.0.2
[conda] blas                      2.109                       mkl    conda-forge
[conda] blas-devel                3.9.0                     9_mkl    conda-forge
[conda] cudatoolkit               11.1.1               heb2d755_7    conda-forge
[conda] libblas                   3.9.0                     9_mkl    conda-forge
[conda] libcblas                  3.9.0                     9_mkl    conda-forge
[conda] liblapack                 3.9.0                     9_mkl    conda-forge
[conda] liblapacke                3.9.0                     9_mkl    conda-forge
[conda] mkl                       2021.2.0           hb70f87d_389    conda-forge
[conda] mkl-devel                 2021.2.0           h57928b3_390    conda-forge
[conda] mkl-include               2021.2.0           hb70f87d_389    conda-forge
[conda] numpy                     1.21.0           py39h6635163_0    conda-forge
[conda] pytorch                   1.9.0           py3.9_cuda11.1_cudnn8_0    pytorch
[conda] pytorch-pretrained-bert   0.6.2                    pypi_0    pypi
[conda] torchaudio                0.9.0                      py39    pytorch
[conda] torchinfo                 0.1.5                    pypi_0    pypi
[conda] torchtext                 0.10.0                     py39    pytorch
[conda] torchvision               0.10.0               py39_cu111    pytorch
[conda] torchviz                  0.0.2                    pypi_0    pypi

## Additional context

I'm not sure why it seems like I'm the only one experiencing this issue. I googled it extensively, but the only similar problems I found were when people were outputting tensorboard information on the same folder from different python scripts; something that is unrelated to my issue.

Perhaps I'm doing something wrong here?
